### PR TITLE
Avoid YAML deprecation

### LINKED ISSUE #4597: yaml.load() deprecation warning after updating to pyyaml v5.1 (released 13/03/19)
Very small issue. Wanted to flag early. Just updated to the lastest version of pyyaml (a dependency of dask) v5.1. After updating I get the following deprecation warning. 

```
/dask/config.py:168: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(f.read()) or {}
```

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #4597

https://github.com/yaml/pyyaml/wiki/PyYAML-yaml.load(input)-Deprecation

- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
Once this is in and released I suggest that we point all other repositories towards `dask.config.yaml_load`

**Comment by mrocklin** [comment]
Merging in 24 hours if there are no comments

File: dask/config.py [hunk 1]
```diff
@@ -6,8 +6,13 @@
 import threading
 try:
     import yaml
+    from distutils.version import LooseVersion
+    if yaml.__version__ >= LooseVersion('5.1'):
+        yaml_load = yaml.safe_load
```
Comment (jakirkham) [inline dask/config.py]: Should we just always use `safe_load`? It's been around for a while.

**Comment by mrocklin** [comment]
Oh great.  Yes, that sounds much nicer.  Fixed.

**Comment by jakirkham** [comment]
Thanks Matt :)

**Pull request merged**